crypto/init.c

```diff
@@ -307,7 +307,7 @@ DEFINE_RUN_ONCE_STATIC(ossl_init_engine_openssl)
     engine_load_openssl_int();
     return 1;
 }
-# ifndef OPENSSL_NO_DEVCRYPTOENG
+# if !defined(OPENSSL_NO_DEVCRYPTOENG) && !defined(OPENSSL_DEVCRYPTO_DYNAMIC)
```

Comment:
Following the logic of "no" macros, this should be `#if !defined(OPENSSL_NO_DEVCRYPTOENG) || defined(OPENSSL_NO_DEVCRYPTOENG_DYNAMIC)`

Reply:
I'll disobey you on this one :wink:.  Code should execute if the static engine is to be loaded: !defined(`OPENSSL_NO_DEVCRYPTOENG) && defined(OPENSSL_NO_DEVCRYPTOENG_DYNAMIC`).

Reply:
D'oh!  Of course...